### PR TITLE
Add libbpf package

### PR DESCRIPTION
**Pull request by ndbaker1**
**Issue number:**

Closes #

**Description of changes:**

Add `libbpf` package for bpf development

**Testing done:**

build kits with `make` and upstream source fallback enabled

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: packages/libbpf/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+releases-url = "https://github.com/libbpf/libbpf/releases"
+
+[[package.metadata.build-package.external-files]]
+url = "https://github.com/libbpf/libbpf/archive/refs/tags/v1.4.3.tar.gz"
+sha512 = "d03965f8abeaac840cda28fdacf9df378869fa6e5ef45d992fc5398bed5970ebedb24f20676b1779f8941b4b912bb691948e198eff87ea1d5d8e5cd95d6e06d1"
+
+# RPM BuildRequires
+[build-dependencies]
+glibc = { path = "../glibc" }
+libelf = { path = "../libelf" }
```
Comment (bcressey) [inline packages/libbpf/Cargo.toml:21]: nit: to make it clear that `libbpf` also depends on zlib:
```suggestion
libelf = { path = "../libelf" }
libz = { path = "../libz" }
```

File: packages/libbpf/libbpf.spec [hunk 2]
```diff
@@ -0,0 +1,51 @@
+Name: %{_cross_os}libbpf
+Version: 1.4.3
+Release: 1%{?dist}
+Summary: Library for BPF
+License: LGPL-2.1 OR BSD-2-Clause
+URL: https://github.com/libbpf/libbpf
+Source0: https://github.com/libbpf/libbpf/archive/refs/tags/v%{version}.tar.gz
+BuildRequires: %{_cross_os}glibc-devel
+BuildRequires: %{_cross_os}libelf-devel
```
Comment (bcressey) [inline packages/libbpf/libbpf.spec:9]:
```suggestion
BuildRequires: %{_cross_os}libelf-devel
BuildRequires: %{_cross_os}libz-devel
```

File: packages/libbpf/libbpf.spec [hunk 3]
```diff
@@ -0,0 +1,51 @@
+Name: %{_cross_os}libbpf
+Version: 1.4.3
+Release: 1%{?dist}
+Summary: Library for BPF
+License: LGPL-2.1 OR BSD-2-Clause
```
Comment (bcressey) [inline packages/libbpf/libbpf.spec]: `LGPL-2.1` is deprecated as an identifier:
```suggestion
License: LGPL-2.1-only OR BSD-2-Clause
```

**Review by bcressey — APPROVED** [review]
LGTM, just a couple minor tweaks.

File: packages/libbpf/libbpf.spec (hunk 3 above)
Comment (ndbaker1) [reply packages/libbpf/libbpf.spec]: ah thanks for this catch!

**Review by yeazelm — APPROVED** [review]

File: packages/libbpf/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+[lib]
+path = "../packages.rs"
+
+[package.metadata.build-package]
+releases-url = "https://github.com/libbpf/libbpf/releases"
+
+[[package.metadata.build-package.external-files]]
+url = "https://github.com/libbpf/libbpf/archive/refs/tags/v1.4.3.tar.gz"
+sha512 = "d03965f8abeaac840cda28fdacf9df378869fa6e5ef45d992fc5398bed5970ebedb24f20676b1779f8941b4b912bb691948e198eff87ea1d5d8e5cd95d6e06d1"
```
Comment (monirul) [inline packages/libbpf/Cargo.toml:16]: Is this SHA published anywhere in the official sites or we generated it from the file after downloading it?

**Review by monirul — APPROVED** [review]
LGTM

File: packages/libbpf/Cargo.toml (hunk 4 above)
Comment (ndbaker1) [reply packages/libbpf/Cargo.toml:16]: generated it after downloading the file, not published afaict

**Review by arnaldo2792 — APPROVED** [review]

**Pull request merged**